lib/dialects/mssql/query.js

```diff
@@ -315,6 +315,16 @@ class Query extends AbstractQuery {
       id = id || (results && results[0][autoIncrementField]);
       id = id || (autoIncrementFieldAlias && results && results[0][autoIncrementFieldAlias]);
 
+      const dataValues = this.instance.dataValues;
+      for (let key in dataValues) {
+        if (dataValues.hasOwnProperty(key) && dataValues[key]) {
+          if (dataValues[key].constructor.name === 'literal') {
```

Comment:
@felixfbecker I will follow @janmeier suggestion and map the whole result back to the instance - will commit soon with the requested changes